File: reversefactorial.go
package main

import (
	"bufio"
	"fmt"
	"os"
    "strconv"
)

func reverse_factorial(text string) string {
    x, _ := strconv.Atoi(text)
    i := 1
    for x > 1 {
        x = x / i
        i += 1
        if x == 1 {
            return text + " = " + strconv.Itoa(i - 1) + "!"
        }
    }
    return text + " NONE"
}

func main() {
	fileHandle, _ := os.Open("reversefactorial_input.txt")
	defer fileHandle.Close()
	fileScanner := bufio.NewScanner(fileHandle)
	for fileScanner.Scan() {
        fmt.Println(reverse_factorial(fileScanner.Text()))
	}
}

File: reversefactorial.py
for line in open('reversefactorial_input.txt', 'r').readlines():
    x = int(line.strip())
    i = x / x
    while x > 1:
        x /= i
        i += 1
        if x == 1:
            print '{} = {}!'.format(line.strip(), i - 1)
            break
    else:
        print '{} NONE'.format(line.strip())
